### PR TITLE
[CPU] Add native support for Qwen3-next

### PR DESCRIPTION
**Pull request by blzheng**
## Motivation

This pr adds native support for Qwen3-next on CPU.

## Modifications

1. add CPU native implementations for the following operations:
  a.  `causal_conv1d_fn`
  b.  `causal_conv1d_update`
  c.  `chunk_gated_delta_rule`
  d.  `fused_sigmoid_gating_delta_rule_update`
  e.  `fused_gdn_gating`
  f.  `Qwen3NextRMSNormGated`
3. fix issues in amx backend:
  a. Weight packing dtype check: weight packing did not support torch.float. This pr adds dtype validation before packing weight
  b. HybridLinearKVPool layer ID handling: Only full attention layers can access get_value_buffer, but layer_id = 0 is not always a full attention layer. This PR updates the logic to handle such cases correctly.
  c. Top-k kernel support: Top-k related kernels lacked support for num_experts = 512. This PR adds support for this configuration.

## Accuracy Tests

Accuracy on GSM8k:
command line: `SGLANG_USE_CPU_ENGINE=1 python -m sglang.launch_server  --model Qwen/Qwen3-Next-80B-A3B-Instruct --trust-remote-code --device cpu --tp 4 --dtype bfloat16   --mem-fraction-static 0.8 --max-total-tokens 65536  --disable-overlap-schedule`
Accuracy: 0.942
Invalid: 0.000
Latency: 3855.785 s
Output throughput: 42.622 token/s

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/layers/attention/mamba/causal_conv1d.py [hunk 1]
```diff
@@ -127,3 +128,109 @@ def causal_conv1d_update(
...
+    conv_states: Optional[torch.Tensor] = None,
+    activation: Optional[str] = "silu",
+    pad_slot_id: int = PAD_SLOT_ID,
+    **kwargs,
+):
+    batch_size = query_start_loc.shape[0] - 1
+    start_q = 0
+    for i in range(batch_size):
+        end_q = query_start_loc[i + 1]
+        x_i, final_states = causal_conv1d_ref(
```
Comment (mingfeima) [inline python/sglang/srt/layers/attention/mamba/causal_conv1d.py:182]: let's wait until sgl-kernel is merged and then replace all the `ref` with real kernels.

File: python/sglang/srt/layers/layernorm.py [hunk 2]
```diff
@@ -463,6 +476,24 @@ def extra_repr(self):
...
 
 
+class Qwen3NextRMSNormGated(nn.Module):
+    def __init__(self, hidden_size, eps=1e-6, **kwargs):
+        super().__init__()
+        self.weight = nn.Parameter(torch.ones(hidden_size))
+        self.variance_epsilon = eps
+
+    def forward(self, hidden_states, gate=None):
+        input_dtype = hidden_states.dtype
```
Comment (yizhang2077) [inline python/sglang/srt/layers/layernorm.py:486]: can we directly use `forward_cpu` in  `RMSNorm`?

File: sgl-kernel/csrc/cpu/topk.cpp [hunk 3]
```diff
@@ -525,6 +528,9 @@ topk_softmax_cpu(at::Tensor& hidden_states, at::Tensor& gating_output, int64_t t
       case 256:
         LAUNCH_TOPK_SOFTMAX_KERNEL(256);
         break;
+      case 512:
+        LAUNCH_TOPK_SOFTMAX_KERNEL(512);
```
Comment (yizhang2077) [inline sgl-kernel/csrc/cpu/topk.cpp:532]: we need split it into another pr
